### PR TITLE
Plot layer features

### LINKED ISSUE #99: Automated widget creation from measured data
It would be cool to have a function that eats an instance of our custom measurements results class (see #98 ) and creates a measurement display widget of the indicated type (line plot/histogram/etc) - or a QToolBox Stack of such widgets.

This widget should have a few functions:
- export data to csv
- automatically update if slider is moved through time
- batch-measure all timepoints

### PR DESCRIPTION
**Pull request by zoccoler**
Hi Johannes @jo-mueller ,

This is my current implementation of the histogram of features, which could fix #99 .
Let me know if it works and if you have suggestions to improve it! As you suggested, I based it a lot on [napari-matplotlib.FeatureScatterWidget](https://github.com/matplotlib/napari-matplotlib/blob/308134b6dc568d88b7ac81d83fba25af8d2ed6fa/src/napari_matplotlib/scatter.py#L111).

Best,
Marcelo

### CODE REVIEW

File: src/napari_stress/napari.yaml [hunk 1]
```diff
@@ -43,6 +43,11 @@ contributions:
     - id: napari-stress.fit_ellipsoid_vectors
       python_name: napari_stress._surface:fit_ellipsoid_to_pointcloud_vectors
       title: Fit PCA ellipsoid to pointcloud and return major axis
+      
+    # Plotting
+    - id: napari-stress.features_histogram
+      python_name: napari_stress:FeaturesHistogramWidget
+      title: Features Histogram
```
Comment (jo-mueller) [inline src/napari_stress/napari.yaml:50]:
```suggestion
      title: Visualize features (histogram)
```

File: src/napari_stress/napari.yaml [hunk 2]
```diff
@@ -84,6 +89,10 @@ contributions:
     - command: napari-stress.fit_ellipsoid_vectors
       autogenerate: true
       display_name: Fit ellipsoid to pointcloud (major axis)
+      
+    # Plotting
+    - command: napari-stress.features_histogram
+      display_name: Features Histogram
```
Comment (jo-mueller) [inline src/napari_stress/napari.yaml:95]:
```suggestion
      display_name: Visualize features (histogram)
```

**Review by jo-mueller — APPROVED** [review]
Hi @zoccoler ,

I love it :) I just changed the naming of the plugin a bit - further down the road there may be more visualization widgets and then it would be nice to have them grouped nicely by name. Thanks a lot for working on this!

**Comment by jo-mueller** [comment]
Maybe as an addon - would it be difficult to add an `Export` button to the plugin? I guess the export function should then open a [FileSelectionDialogue](https://doc.qt.io/qtforpython-5/PySide2/QtWidgets/QFileDialog.html#PySide2.QtWidgets.PySide2.QtWidgets.QFileDialog.getSaveFileName) and call the `dataframe.to_csv()` function on that.

**Review by jo-mueller — CHANGES_REQUESTED** [review]
Maybe as an addon - would it be difficult to add an `Export` button to the plugin? I guess the export function should then open a [FileSelectionDialogue](https://doc.qt.io/qtforpython-5/PySide2/QtWidgets/QFileDialog.html#PySide2.QtWidgets.PySide2.QtWidgets.QFileDialog.getSaveFileName) and call the `dataframe.to_csv()` function on that.

File: src/napari_stress/_visualization/features_histogram.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        napari.layers.Tracks,
+        napari.layers.Vectors,
+    )
+
+    def __init__(self, napari_viewer: napari.viewer.Viewer):
+        super().__init__(napari_viewer)
+        self._key_selection_widget = magicgui(
+            self._set_axis_keys,
+            x_axis_key={"choices": self._get_valid_axis_keys},
+            n_bins={"value": 100, "widget_type": "SpinBox"},
```
Comment (jo-mueller) [inline src/napari_stress/_visualization/features_histogram.py:32]:
```suggestion
            n_bins={"value": 50, "widget_type": "SpinBox"},
```

Comment (jo-mueller) [reply src/napari_stress/_visualization/features_histogram.py:32]: Just a matter of taste - feel free to also leave it as it is.

File: src/napari_stress/_visualization/features_histogram.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+            call_button="plot",
```
Comment (jo-mueller) [inline src/napari_stress/_visualization/features_histogram.py:33]:
```suggestion
            call_button="Update",
```

**Comment by zoccoler** [comment]
> Maybe as an addon - would it be difficult to add an `Export` button to the plugin? I guess the export function should then open a [FileSelectionDialogue](https://doc.qt.io/qtforpython-5/PySide2/QtWidgets/QFileDialog.html#PySide2.QtWidgets.PySide2.QtWidgets.QFileDialog.getSaveFileName) and call the `dataframe.to_csv()` function on that.

I implemented that now, could you give it a try? @jo-mueller . It saves the data displayed on the graph, not the whole data in `layer.features`. It made more sense to me since it comes along with the plotting widget, but we can change that later if you like or put this option somewhere else.

Default place to save is 'c:\\', it works in Windows, I don't know for other OS.

**Review by jo-mueller — APPROVED** [review]
Works like a charm! 

One tiny thing: Could you remove the `LBDV_at_Quad_Pts_...` files from the PR? They are generated on-the-fly by some under-the-hood stress code and I haven't yet added them to the gitignore ^^"

Feel free to merge.

**Pull request merged**